app/views/keywords/_keyword.html.erb

```diff
@@ -1,17 +1,17 @@
 <div id="data-container-<%= keyword.id %>">
-  <% if (defined?(status) && status.eql?('complete')) || keyword.search_result.present? %>
+  <% if keyword.complete? %>
     <li class="list-group-item">
       <%= keyword.name %> - <%= link_to 'View Result', search_result_keyword_path(keyword.id), target: '_blank' %>
     </li>
-  <% elsif defined?(status) && status.eql?('failed') %>
+  <% elsif keyword.failed? %>
     <li class="list-group-item">
       <%= keyword.name %> - (An Error Occurred, Will retry soon..)
     </li>
-  <% elsif defined?(status) && status.eql?('searching') %>
+  <% elsif keyword.searching? %>
     <li class="list-group-item">
       <%= keyword.name %> - (Please Wait, Searching...)
     </li>
-  <% elsif defined?(status) && status.eql?('retrying') %>
+  <% elsif keyword.retrying? %>
```

Comment:
To simplify/DRYup views, presenters can be used (see https://medium.com/@steve.v.brewer/rails-presenters-for-readable-modular-and-testable-code-e63431360c66):

 ```
<li class="list-group-item">
  <%= keyword_presenter.display_text %> - <%= keyword_presenter.status_info %>
</li>
```

Then, in `app/presenters/keyword_presenter.rb`:

```ruby
class KeywordPresnter
  def initialize(keyword)
     @keyword = keyword
  end

  def display_text
     @keyword.name
  end 

  def status_info
     case(@keyword.search_status) do
        when 'searching'
           '(Please Wait, Searching...)'
        #...
        breakl
     end 
  end
end
```